### PR TITLE
Private views

### PR DESCRIPTION
**Pull request by bnmajor**
This PR contains a lot of new features and behaviors:

- Auto-save a default View
  - Once a parameter is added the auto-save feature is enabled
  - The last added parameter is used to determine which simulation and run to associate the auto-saved (i.e. "default") View with
  - If a user navigates to a run with an associated default view they will shown a dialog that allows them to choose whether or not to load that default view
  - When a user navigates to a new run the auto-save will stop until a new parameter is added again. This is to prevent the default view from being associated with the wrong simulation/run.
  - Text has been added to the bottom of the view controls to indicate that the View is being auto-saved and when the last save happened
    autosave_text
- Remember time step
  - When a View is save (automatically or manually) the current time step at the time of save is saved with the View
  - When a View is loaded it will automatically be loaded at the step associated with it
- Improved UI
  - Cleaned up sizing of some of the dialog boxes so that text is not wrapped mid-word
  - Add a tab to LoadView dialog to filter between user Views and all available Views
  - Add an icon to the LoadView dialog table that allows users to toggle the public/private status of their View(s)
- New View name requirements relaxed
  - Users can use any name that they would like for their View
  - If that user has an existing View with that name they are warned in the SaveView dialog and prompted to confirm that they wish to overwrite the existing View on save
- Load View as template
  - If a user opens the LoadView dialog within a run they will be able to load some Views as a template
  - Views qualify to be loaded as a template if they share the same simulation but *not* the same run as the current run.
  - If a template contains parameters not available within the current run the cell will simply be left empty
  - In the case of templates the time step associated with the View is ignored and is instead reset to step 1

### CODE REVIEW

File: client/src/components/core/App/script.js [hunk 1]
```diff
@@ -154,17 +163,23 @@ export default {
 
     initialDataLoaded(num_timesteps, itemId) {
       this.numLoadedGalleries += 1;
-      if (this.dataLoaded) {
-        return;
-      }
+      if (this.dataLoaded)
+          return
```
Comment (cjh1) [inline client/src/components/core/App/script.js:167]: Why this change? Usually its better to keep the braces as it avoids issues with added another line to the if block without adding the braces ....

**Review by cjh1 — APPROVED** [review]
LGTM

File: client/src/components/core/App/script.js (hunk 1 above)
Comment (bnmajor) [reply client/src/components/core/App/script.js:167]: No good reason, just bad habit. I've fixed this now.
